### PR TITLE
Add monitoring and logging to datenlord

### PR DESCRIPTION
**Pull request by mrwwheat7**

### CODE REVIEW

File: scripts/datenlord-deploy.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+if [ -z "$1" ]
+then
+    kubectl apply -f datenlord-logging.yaml
+    kubectl apply -f datenlord-monitor.yaml
+else
```
Comment (pzheng1025) [inline scripts/datenlord-deploy.sh:7]: define a valid value for $1?

File: scripts/datenlord-deploy.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    helm repo add stable https://charts.helm.sh/stable
+    helm install prometheus stable/prometheus-operator --namespace monitoring
+    helm repo add elastic https://helm.elastic.co
+    helm install filebeat elastic/filebeat
+    helm install kibana elastic/kibana
+    helm install elasticsearch elastic/elasticsearch -f values.yaml
+fi
```
Comment (pzheng1025) [inline scripts/datenlord-deploy.sh:14]: Maybe change the name of this script to deploy-monitoring or others. It doesn't deploy DatenLord.

File: scripts/datenlord-monitor.yaml [hunk 3]
```diff
@@ -0,0 +1,740 @@
...
+      prometheus.io/scrape: 'true'
+      prometheus.io/port:   '3000'
+spec:
+  selector: 
+    app: grafana
+  type: NodePort  
+  ports:
+    - port: 3000
+      targetPort: 3000
+      nodePort: 32000
```
Comment (pwang7) [inline scripts/datenlord-monitor.yaml:740]: add extra empty line

File: scripts/datenlord-monitor-deploy.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    helm repo add stable https://charts.helm.sh/stable
+    helm install prometheus stable/prometheus-operator --namespace datenlord-monitoring 
+    helm repo add elastic https://helm.elastic.co
+    helm install filebeat elastic/filebeat
+    helm install kibana elastic/kibana
+    helm install elasticsearch elastic/elasticsearch -f values.yaml
+else
+    kubectl apply -f datenlord-logging.yaml
+    kubectl apply -f datenlord-monitor.yaml
+fi
```
Comment (pwang7) [inline scripts/datenlord-monitor-deploy.sh:15]: add extra empty line

File: scripts/Datenlord_Dashboard.json [hunk 5]
```diff
@@ -0,0 +1,3831 @@
...
+      "2d",
+      "7d",
+      "30d"
+    ]
+  },
+  "timezone": "browser",
+  "title": "Datenlord Dashboard",
+  "uid": "RPWgg6z7z",
+  "version": 9
+}
```
Comment (pwang7) [inline scripts/Datenlord_Dashboard.json:3831]: add extra empty line

File: .github/workflows/cron.yml [hunk 6]
```diff
@@ -13,17 +13,25 @@ env:
...
+  ELASTICSEARCH_LABEL: elasticsearch
   FUSE_CONTAINER_NAME: datenlord-fuse
   FUSE_MOUNT_PATH: /var/opt/datenlord-data
+  GRAFANA_LABEL: grafana
   K8S_CONFIG: k8s.e2e.config
   K8S_VERSION: v1.21.1
+  KIBANA_LABEL: kibana
   KIND_NODE_VERSION: kindest/node:v1.21.1@sha256:fae9a58f17f18f06aeac9772ca8b5ac680ebbed985e266f711d936e91d113bad
   KIND_VERSION: v0.11.0
+  MINIKUBE_VERSION: v1.13.0
```
Comment (pwang7) [inline .github/workflows/cron.yml:31]: comment out minikube version, since we don't use it

**Review by pwang7 — CHANGES_REQUESTED** [review]

**Review by mrwwheat7 — COMMENTED** [review]
Add test for AlertManager

File: scripts/datenlord-monitor-deploy.sh [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+if [ "$1" == "helm" ]
+then
+    kubectl create namespace datenlord-monitoring 
+    helm repo add stable https://charts.helm.sh/stable
+    helm install prometheus stable/prometheus-operator --namespace datenlord-monitoring 
+    helm repo add elastic https://helm.elastic.co
+    helm install filebeat elastic/filebeat
+    helm install kibana elastic/kibana
+    helm install elasticsearch elastic/elasticsearch -f values.yaml
+    kubectl wait --for=condition=Ready pod -l app.kubernetes.io/name=grafana -n datenlord-monitoring --timeout=60s
```
Comment (mrwwheat7) [inline scripts/datenlord-monitor-deploy.sh:16]: Add status check for promethues

**Review by mrwwheat7 — COMMENTED** [review]
Remove data origin for promethues

File: scripts/datenlord-monitor-deploy.sh [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    kubectl port-forward $POD_NAME 3000 -n datenlord-monitoring
+else
+    kubectl apply -f datenlord-logging.yaml
+    kubectl apply -f datenlord-monitor.yaml
+    kubectl wait --for=condition=Ready pod -l app=prometheus-server -n datenlord-monitoring --timeout=60s
+    kubectl wait --for=condition=Ready pod -l app=grafana -n datenlord-monitoring --timeout=60s
+    kubectl wait --for=condition=Ready pod -l app=kibana -n datenlord-logging --timeout=120s
+    kubectl wait --for=condition=Ready pod -l app=elasticsearch -n datenlord-logging --timeout=120s
+    POD_NAME=`kubectl get pods -l app=grafana -n datenlord-monitoring | grep grafana | awk '{print $1}'`
+    kubectl port-forward $POD_NAME 3000 -n datenlord-monitoring
```
Comment (mrwwheat7) [inline scripts/datenlord-monitor-deploy.sh:29]: Make the port as env variable

File: .github/workflows/cron.yml [hunk 9]
```diff
@@ -244,6 +253,21 @@ jobs:
           sleep 60
           kubectl get pods -A -o wide
           # sudo netstat -lntp && ls -lsh
+      - name: Datenlord Monitoring Test
+        run: |
+          kubectl apply -f $DATENLORD_MONITORING
```
Comment (mrwwheat7) [inline .github/workflows/cron.yml:258]: Change it to datenlord-monitoring-deploy.sh

File: scripts/datenlord-monitor.yaml [hunk 10]
```diff
@@ -0,0 +1,729 @@
...
+      scrape_interval: 5s
+      evaluation_interval: 5s
+    rule_files:
+      - alertmanager_alerts.yaml
+    alerting:
+      alertmanagers:
+      - scheme: http
+        static_configs:
+        - targets:
+          - "alertmanager.datenlord-monitoring.svc:8080"
```
Comment (mrwwheat7) [inline scripts/datenlord-monitor.yaml:428]: Extract static configs to another ConfigMap, and dynamically config it.

File: .github/workflows/cron.yml [hunk 11]
```diff
@@ -204,6 +213,18 @@ jobs:
           rm -rf $HOME/.ssh/
           ssh-keygen -N '' -f ~/.ssh/id_rsa
           /bin/bash /tmp/setup.sh
+      - name: Datenlord Monitoring Test
+        run: |
+          sudo chmod 755 scripts/datenlord-monitor-deploy.sh
+          cat scripts/alertmanager_test_alert.yaml >> scripts/alertmanager_alerts.yaml
+          sh scripts/datenlord-monitor-deploy.sh deploy
+          FOUND_PATH=`curl --silent prometheus-service.datenlord-monitoring.svc:8080 | grep Found | awk '{print $1}'`
+          test -n #FOUND_PATH || (echo "FAILED TO FIND PROMETHEUS SERVICE" && /bin/false)
```
Comment (pzheng1025) [inline .github/workflows/cron.yml:222]: $FOUND_PATH ?

File: scripts/values.yaml [hunk 12]
```diff
@@ -0,0 +1,30 @@
+---
```
Comment (pzheng1025) [inline scripts/values.yaml:1]: Could you change the file name to give more information? like xxx_value.yaml?

**Review by pzheng1025 — APPROVED** [review]

**Review by pwang7 — APPROVED** [review]

**Pull request merged**